src/health.ts

```diff
@@ -0,0 +1,55 @@
+import fetch from 'node-fetch';
+import { AbortSignal } from 'node-fetch/externals';
+import { KubeConfig } from './config';
+import { RequestOptions } from 'node:https';
+
+export class Health {
+    public config: KubeConfig;
+
+    public constructor(config: KubeConfig) {
+        this.config = config;
+    }
+
+    public async readyz(opts: RequestOptions): Promise<boolean> {
+        return this.check('/readyz', opts);
+    }
+
+    public async livez(opts: RequestOptions): Promise<boolean> {
+        return this.check('/livez', opts);
+    }
+
+    private async healthz(opts: RequestOptions): Promise<boolean> {
+        return this.check('/healthz', opts);
+    }
+
+    private async check(path: string, opts: RequestOptions): Promise<boolean> {
+        const cluster = this.config.getCurrentCluster();
+        if (!cluster) {
+            throw new Error('No currently active cluster');
+        }
+
+        const requestURL = new URL(cluster.server + path);
+        const requestInit = await this.config.applyToFetchOptions(opts);
+        const controller = new AbortController();
+        requestInit.signal = controller.signal as AbortSignal;
+        requestInit.method = 'GET';
+
+        try {
+            const response = await fetch(requestURL.toString(), requestInit);
+            const status = response.status;
+            if (status === 200) {
+                return true;
+            }
+            if (status === 404) {
+                if (path === '/healthz') {
+                    // /livez/readyz return 404 and healthz also returns 404, let's consider it is live
+                    return true;
+                }
+                return this.healthz(opts);
+            }
+            return false;
+        } catch {
+            throw new Error('Error occurred in health request');
```

Comment:
You can attach the original error like this, while still throwing a custom error.

```suggestion
        } catch (err) {
            throw new Error('Error occurred in health request', { cause: err });
```

Reply:
Changing the tsconfig with the following would fix the problem, but I'm not sure of the potential side-effects:

```
  "compilerOptions": {
    "lib": ["es2022"],
```

Reply:
Ah, ok. There must be a TypeScript setting that needs to be tweaked in the project since [`cause` is a newer addition to the spec](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause). I will resolve my comment.